app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,55 @@
+class ReviewsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+
+  def show
+    @review = Review.find(params[:id])
+    @pizzeria = Pizzeria.find(@review.pizzeria_id)
+  end
+
+  def new
+    @pizzeria = Pizzeria.find(params[:pizzeria_id])
+    @review = Review.new
+  end
+
+  def create
+    @pizzeria = Pizzeria.find(params[:pizzeria_id])
+    @review = @pizzeria.reviews.build(review_params)
+    @review.user = current_user
+    if @review.save
+      flash[:notice] = "Review created"
+      redirect_to pizzeria_review_path(@pizzeria, @review)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @review = Review.find(params[:id])
+    @pizzeria = Pizzeria.find(@review.pizzeria_id)
+    if !correct_user?
+      flash[:error] = "You do not have permission to edit this review"
+      render :show
+    end
+  end
+
+  def update
+    @review = Review.find(params[:id])
```

Comment:
You can avoid having to check the user explicitly by scoping this `find` to just search the user's own reviews:

```ruby
@review = current_user.reviews.find(params[:id])
```

This will raise an `ActiveRecord::NotFound` exception if a user tries to update a review they didn't create (and Rails will return a `404 Not Found` status code).